server/channels/jobs/batch_report_worker.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package jobs
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+	"github.com/pkg/errors"
+)
+
+type BatchReportWorkerAppIFace interface {
+	SaveReportChunk(format string, prefix string, count int, reportData []model.ReportableObject) *model.AppError
+	CompileReportChunks(format string, prefix string, numberOfChunks int, headers []string) (string, *model.AppError)
+	SendReportToUser(userID string, filename string) *model.AppError
+}
+
+type BatchReportWorker[T BatchReportWorkerAppIFace] struct {
+	BatchWorker[T]
+	reportFormat string
+	headers      []string
+	getData      func(jobData model.StringMap, app T) ([]model.ReportableObject, model.StringMap, bool, error)
+}
+
+func MakeBatchReportWorker[T BatchReportWorkerAppIFace](
```

Comment:
A reporting job will create this worker, specifying the report format and the headers to use once compiled.